renderdoc/os/posix/android/android_stringio.cpp

```diff
@@ -63,15 +64,33 @@ namespace FileIO
 {
 const char *GetTempRootPath()
 {
-  return "/sdcard";
+  static string ret;
+  GetExecutableFilename(ret);
+
+  // This folder is writable even if the APK does not have manifest write permissions.
+  ret = "/data/data/" + ret + "/files";
+  return ret.c_str();
 }
 
+// For RenderDocCmd.apk, this returns "org.renderdoc.renderdoccmd"
+// For other APKs, we use it to get the writable temp directory.
 void GetExecutableFilename(string &selfName)
 {
-  char path[512] = {0};
-  readlink("/proc/self/exe", path, 511);
+  char buf[4096];
+  snprintf(buf, sizeof(buf), "/proc/%u/cmdline", getpid());
```

Comment:
On android is it not possible to pass command line arguments? So this is equivalent to `argv[0]`?